browser/android/background_video/background_video_playback_tab_helper.cc

```diff
@@ -65,15 +113,21 @@ BackgroundVideoPlaybackTabHelper::BackgroundVideoPlaybackTabHelper(
 
 BackgroundVideoPlaybackTabHelper::~BackgroundVideoPlaybackTabHelper() {}
 
-void BackgroundVideoPlaybackTabHelper::DidFinishNavigation(
-    content::NavigationHandle* navigation_handle) {
+void BackgroundVideoPlaybackTabHelper::PrimaryMainDocumentElementAvailable() {
+  content::WebContents* contents = web_contents();
   // Filter only YT domain here
-  if (!IsYouTubeDomain(web_contents()->GetLastCommittedURL())) {
+  if (!IsYouTubeDomain(contents->GetLastCommittedURL())) {
     return;
   }
-  if (IsBackgroundVideoPlaybackEnabled(web_contents())) {
-    web_contents()->GetPrimaryMainFrame()->ExecuteJavaScript(
-        kYoutubeBackgroundPlaybackScript, base::NullCallback());
+  content::RenderFrameHost::AllowInjectingJavaScript();
+  if (IsBackgroundVideoPlaybackEnabled(contents)) {
+    contents->GetPrimaryMainFrame()->ExecuteJavaScript(
+        kYoutubeBackgroundPlayback, base::NullCallback());
+  }
+  if (base::FeatureList::IsEnabled(
```

Comment:
in a follow-up we need to update the tests to check that the script doesn't get injected when the feature is disabled